ext/webextension/src/browser_action/main_popup.js

```diff
@@ -241,21 +241,54 @@ function updateUIForDomainSettings(combined)
 }
 
 function extractDomainFromUrl(url) {
-    if (!url || url.startsWith('about:')
-        || url.startsWith('resource:')
-        || url.startsWith('moz-extension:')
-        || url.startsWith('chrome-extension:')
-        || url.startsWith('chrome:'))
-        url = '';
-    let domain_parts = parseUri(url).domain.split(".");
-    let significant_parts = 2;
-    const common_slds = ['co','com','gov','govt','net','org','edu','priv','ac'];
-    let second_level_domain = domain_parts[domain_parts.length-2].toLowerCase();
-    if (domain_parts.length > 2 && common_slds.includes(second_level_domain))
-        significant_parts = 3;
-
-    let significant_domain = domain_parts.slice(-significant_parts)
-    return [domain_parts, significant_domain];
+    if (!url)
+        throw "extractDomainFromUrl: No url provided!";
+
+    var urlParsed = parseUri(url);
+    {
+        let protocolsIgnored = ["about", "resource", "moz-extension", "chrome-extension", "chrome", "edge", "brave"];
+        if (protocolsIgnored.includes(urlParsed.protocol)) 
+            throw "extractDomainFromUrl: Invalid url protocol provided";
+    }
+
+    let rxDomainMatchers = [
+        /([^\.]+\.[^\.]+)$/i,   // Default matcher matches two-part domain names (.* pattern) i.e. github.com
+    ];
+    
+    config.treat_as_same_site.split('\n').forEach(function(pattern) {
```

Comment:
I know I'm often guilty of this my self, but hey, now I'm in review mode and trying to understand someone else's code ;) 

Could it be better to make it a named function and do something like
`let matchRegEx = config.treat_as_same_site.split('\n').map(translate_match_string_to_regex);
rxDomainMatchers.push(...matchRegEx.map(pattern => new RegExp(pattern, 'i')))`